src/routes/profile/data.ts

```diff
@@ -1,13 +1,49 @@
 import { DBTables, SCHEMA_VERSION } from '../../constants/db.ts';
-import type { Profile } from './validation.ts';
+import type { CreateProfileData, Profile, UpdateProfileData } from './validation.ts';
 
-export async function getProfileById(database: D1Database, profileId: string) {
+export async function insertProfile(database: D1Database, body: CreateProfileData) {
+	const id = crypto.randomUUID();
+	const insertedAt = new Date().toISOString();
+	const happenedAt = insertedAt;
+
+	const { success } = await database.prepare(`
+		INSERT INTO ${DBTables.PROFILE} (
```

Comment:
Curiosity: We are writing plain SQL here. What made us take this path (is it that our devs are comfortable with it, doing it for learning rather than ORMs like Knex?

Reply:
So, we discussed using a query builder or ORM before...

The argument against ORMs is that in general they are "everything and the kitchen sink", so would lock us to using the ORM's specific way of writing things and bloating the project.

Then the option was using a query builder like Knex, and we were almost doing that, but Cloudflare D1, our db of choice for the project, is not supported by Knex. Other query builders exist but are not as popular.

So, we decided to just go with plain SQL statements and use that as an opportunity for everyone to learn and improve their db skills.

Reply:
@rjtbansal prepared statements so far has been very effective, sufficiently secure and easy to use even though most of the team is used to working with ORM's.